doc/site/README.md

```diff
@@ -1,31 +1,62 @@
-# Recoil Site
+# Recoil Site Readme
 
-The recoil site is a jekyll application, usually hosted on github pages.
+## Setting Up Local Site Development
 
-## Testing locally
+For basic local site development, all that is required is [Hugo](https://gohugo.io/installation/).
 
-Install Ruby 3.2.3 ([newer versions can cause problems](https://stackoverflow.com/a/77896791/317135)).
+## Testing And Modifying Locally
 
-Inside `doc/site` run:
+`cd` into `doc/site` and run `hugo server`. It will take 2-5 seconds to generate the site, but then any changes you make will be compiled quickly.
 
-```bash
-bundle
-bundle exec jekyll build && bundle exec jekyll serve
-```
+It's worth noting that since it's an incremental compile by default, if you start getting odd behavior, restart the server to build it fresh. This can happen a lot if you're messing with partials or layouts.
+
+See [Writing Articles guide](content/development/writing-site-articles.md) for info on writing articles.
+
+## Generating and Testing Lua Docs
+
+By default, `doc/site/content/docs/lua-api` is not tracked in the repo, since they are autogenerated.
+To generate, install the following:
 
-Navigate to http://localhost:4000/spring
+1. [Hugo](https://gohugo.io/installation/)
+2. [Cargo](https://doc.rust-lang.org/cargo/getting-started/installation.html) (for installing doc tool)
+3. [Node](https://nodejs.org/en/download)
+4. [Python 3](https://www.python.org/downloads/)
+5. [emmylua_doc_cli](https://github.com/CppCXY/emmylua-analyzer-rust/tree/main/crates/emmylua_doc_cli)
+6. [lua-doc-extractor](https://github.com/rhys-vdw/lua-doc-extractor) version 3
 
-## Generating Lua API
+You can run commands manually or use [Task](https://taskfile.dev/installation/).
 
-Have [emmylua_doc_cli](https://github.com/CppCXY/emmylua-analyzer-rust/tree/main/crates/emmylua_doc_cli) and [lua-doc-extractor](https://github.com/rhys-vdw/lua-doc-extractor) installed and available in `$PATH`.
+`task generate-docs` will extract lua meta files, generate markdown, and format it into the Hugo website. Can only be called in `doc`.
+Otherwise, do this manually:
 
 ```bash
-rm -rf rts/Lua/library/generated &&
-lua-doc-extractor rts/Lua/*.cpp --dest rts/Lua/library/generated &&
-emmylua_doc_cli \
-  -i rts/Lua/library/ \
-  -o doc/site/lua-api \
-  --override-template doc/emmlua-doc-cli-template/
+lua-doc-extractor rts/Lua/**/*.cpp rts/Rml/SolLua/**/*.cpp --dest rts/Lua/library/generated
```

Comment:
This is actually wrong:

```suggestion
lua-doc-extractor "rts/Lua/**/*.cpp" "rts/Rml/SolLua/**/*.cpp" --dest rts/Lua/library/generated
```

I have added an explanation to lua-doc-extractor: https://github.com/rhys-vdw/lua-doc-extractor/blob/b2159d601ce20015839bb19266dac3df35db21ec/src/cli.ts#L36-L40

Basically if you do `rts/lua/**/*.cpp` it will be expanded by the CLI before being passed, which means glob will not be able to process it. Quoting means the more complex wildcard syntax will be handled by lua-doc-extractor internally (which is correct, it simply won't work without quotes on my linux install anyway).